File: common/src/main/java/common/services/DriverService.java
package common.services;

import common.domain.Driver;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface DriverService {
    List<Driver> findAll();

    List<Driver> findAll(Pageable paging);

    void save(String name, String cnp);

    void update(Long id, String name, String cnp);

    void deleteById(Long id);
}

File: web/src/main/angular-gui/src/app/driver/service/driver.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {DriverDTO} from "../../model/driver";

@Injectable({
  providedIn: 'root'
})
export class DriverService {
  private url = "http://localhost:8085/drivers";

  constructor(private client: HttpClient) { }

  getAll(page: number): Observable<DriverDTO> {
    return this.client.get<DriverDTO>(this.url + `/getAll?page=${page}`);
  }

  addDriver(name: string, cnp: string): Observable<any> {
    return this.client.post(
      this.url + "/save", {
        name: name,
        cnp: cnp
      }
    )
  }

  deleteDriver(id: number): Observable<any> {
    return this.client.delete(this.url + `/delete/${id}`)
  }

  updateDriver(id: number, name: string, cnp: string): Observable<any> {
    return this.client.put(this.url + `/update/${id}`, {
      name: name,
      cnp: cnp
    })
  }
}
